src/test/lambda_examples.rs

```diff
@@ -0,0 +1,5 @@
+#![allow(dead_code)]
+
+#[cfg(feature = "aws-auth")]
+mod auth;
+mod no_auth;
```

Comment:
These are split into two separate modules to avoid including conditional compilation statements within the examples. Although we've talked about containing our examples in the top-level `tests` directory, I needed to include these ones here instead because the `tests` directory does not follow the module structure and therefore does not permit conditional compilation of an entire file.